Modules/IO/MeshBase/include/itkMeshIOTestHelper.h

```diff
@@ -0,0 +1,372 @@
+
+  std::cout << "ComponentTypeAsString: " << meshIO->GetComponentTypeAsString(floatComponent) << std::endl;
+
+  itk::CommonEnums::IOPixel pixelType = itk::CommonEnums::IOPixel::SCALAR;
+  std::cout << "PixelTypeAsString: " << meshIO->GetPixelTypeAsString(pixelType) << std::endl;
+
+  itk::CommonEnums::IOComponent pointComponentType = itk::CommonEnums::IOComponent::FLOAT;
+  meshIO->SetPointComponentType(pointComponentType);
+  ITK_TEST_SET_GET_VALUE(pointComponentType, meshIO->GetPointComponentType());
+
+  itk::CommonEnums::IOComponent cellComponentType = itk::CommonEnums::IOComponent::FLOAT;
+  meshIO->SetCellComponentType(cellComponentType);
+  ITK_TEST_SET_GET_VALUE(cellComponentType, meshIO->GetCellComponentType());
+
+  unsigned int pointDimension = 2;
+  meshIO->SetPointDimension(pointDimension);
+  ITK_TEST_SET_GET_VALUE(pointDimension, meshIO->GetPointDimension());
+
+  itk::MeshIOBase::SizeValueType numberOfPoints = 400;
+  meshIO->SetNumberOfPoints(numberOfPoints);
+  ITK_TEST_SET_GET_VALUE(numberOfPoints, meshIO->GetNumberOfPoints());
+
+  itk::MeshIOBase::SizeValueType numberOfCells = 100;
+  meshIO->SetNumberOfCells(numberOfCells);
+  ITK_TEST_SET_GET_VALUE(numberOfCells, meshIO->GetNumberOfCells());
+
+  itk::MeshIOBase::SizeValueType numberOfPointPixels = 200;
+  meshIO->SetNumberOfPointPixels(numberOfPointPixels);
+  ITK_TEST_SET_GET_VALUE(numberOfPointPixels, meshIO->GetNumberOfPointPixels());
+
+  itk::MeshIOBase::SizeValueType numberOfCellPixels = 600;
+  meshIO->SetNumberOfCellPixels(numberOfCellPixels);
+  ITK_TEST_SET_GET_VALUE(numberOfCellPixels, meshIO->GetNumberOfCellPixels());
+
+  itk::MeshIOBase::SizeValueType cellBufferSize = 1000;
+  meshIO->SetCellBufferSize(cellBufferSize);
+  ITK_TEST_SET_GET_VALUE(cellBufferSize, meshIO->GetCellBufferSize());
+
+  itk::IOFileEnum fileType = itk::IOFileEnum::ASCII;
+  meshIO->SetFileType(fileType);
+  ITK_TEST_SET_GET_VALUE(fileType, meshIO->GetFileType());
+
+  std::cout << "FileTypeAsString: " << meshIO->GetFileTypeAsString(fileType) << std::endl;
+
+  itk::IOByteOrderEnum ioByteOrder = itk::IOByteOrderEnum::BigEndian;
+  meshIO->SetByteOrder(ioByteOrder);
+  ITK_TEST_SET_GET_VALUE(ioByteOrder, meshIO->GetByteOrder());
+
+  std::cout << "ByteOrderAsString: " << meshIO->GetByteOrderAsString(ioByteOrder) << std::endl;
+
+  itk::MeshIOBase::ArrayOfExtensionsType supportedReadExtensions = meshIO->GetSupportedReadExtensions();
+  std::cout << "SupportedReadExtensions: " << std::endl;
+  for (auto ext : supportedReadExtensions)
+  {
+    std::cout << ext << std::endl;
+  }
+
+  itk::MeshIOBase::ArrayOfExtensionsType supportedWriteExtensions = meshIO->GetSupportedWriteExtensions();
+  std::cout << "SupportedWriteExtensions: " << std::endl;
+  for (auto ext : supportedWriteExtensions)
+  {
+    std::cout << ext << std::endl;
+  }
+
+
+  return EXIT_SUCCESS;
+}
+
+void
+AllocateBuffer(void ** data, itk::IOComponentEnum componentType, itk::SizeValueType bufferSize)
```

Comment:
Not sure this is the best thing to do, requiring the caller methods to remember to delete the pointers later. Suggestions are welcome.

Reply:
@PranjalSahu what do you think?

Reply:
Definitely something better is required as the compiler is complaining about the deletion:
```
Modules/IO/MeshBYU/test/itkBYUMeshIOTest.cxx:169:12: warning: deleting 'void*' is undefined [-Wdelete-incomplete]   delete[] pointBuffer
             ^~~~~~~~~~~
```

e.g. https://open.cdash.org/viewBuildError.php?type=1&buildid=7889131